app/test/fuzz_abci_test.go

```diff
@@ -152,6 +202,13 @@ func TestPrepareProposalConsistency(t *testing.T) {
 					// change if PFB transactions are not separated and put into
 					// their own namespace
 					require.GreaterOrEqual(t, len(resp.BlockData.Txs), sendTxCount+1)
+
+					// Differentially test the length of transactions returned by testApp.PrepareProposal.
+					// Essentially, test by extracting the txs from a second data square made with square.Construct
+					// then compare the length of this list and the list returned by testApp.PrepareProposal.
+					// Discussion: https://github.com/celestiaorg/celestia-app/issues/2519
+					reconstrucedTxs := reconstruct(testApp, height, blockTime, coretypes.Txs(txs).ToSliceOfBytes(), encConf.TxConfig)
+					require.Equal(t, len(reconstrucedTxs), len(resp.BlockData.Txs))
```

Comment:
this is great work @exp7l !! its a good thing to test, but we already have tests that reconstruct the block data from the square and compare these things. per this comment I think that what we want is to compare the input to the square with the resulting txs from PrepareProposal